sdk/program/src/alt_bn128_compression.rs

```diff
@@ -0,0 +1,497 @@
+pub mod prelude {
+    pub use crate::alt_bn128_compression::{
+        alt_bn128_compression_size::*, consts::*, target_arch::*, AltBn128CompressionError,
+    };
+}
+
+use thiserror::Error;
+
+mod consts {
+    pub const ALT_BN128_COMPRESSION_G1_COMPRESS: u64 = 0;
+    pub const ALT_BN128_COMPRESSION_G1_DECOMPRESS: u64 = 1;
+    pub const ALT_BN128_COMPRESSION_G2_COMPRESS: u64 = 2;
+    pub const ALT_BN128_COMPRESSION_G2_DECOMPRESS: u64 = 3;
+}
+
+mod alt_bn128_compression_size {
+    pub const G1: usize = 64;
+    pub const G2: usize = 128;
+    pub const G1_COMPRESSED: usize = 32;
+    pub const G2_COMPRESSED: usize = 64;
+}
+
+#[derive(Debug, Error, Clone, PartialEq, Eq)]
+pub enum AltBn128CompressionError {
+    #[error("Unexpected error")]
+    UnexpectedError,
+    #[error("Failed to decompress g1")]
+    DecompressingG1Failed,
```

Comment:
Nit: This is really minor, but can we name it `G1DecompressionFailed` instead to make it slightly more natural? Same with the error variants below.

Reply:
agreed, done